app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(
+            self,
+            name: str,
+            age: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    Person.people = {person_info["name"]: Person(person_info["name"],
+                                                 person_info["age"])
+                     for person_info in people}
+    for person_info in people:
+        person_instance = Person.people[person_info["name"]]
+        if "wife" in person_info and \
+                (wife_name := person_info["wife"]):
```

Comment:
Use `get()` to avoid checking for the presence of the key in the `person_info`, the loop will not enter if condition in case value is absent by this key
```suggestion
        if wife_name := person_info.get("wife"):
```

Reply:
this doesnt work